app/src/main/java/protect/card_locker/ManageGroupsActivity.java

```diff
@@ -228,20 +228,23 @@ public void onMoveUpButtonClicked(View view) {
     @Override
     public void onEditButtonClicked(View view) {
         Intent intent = new Intent(this, ManageGroupActivity.class);
-        intent.putExtra("group", getGroupName(view));
+        final String groupName = getGroupName(view);
+        Group group = DBHelper.getGroupByName(mDatabase, groupName);
+        intent.putExtra("group", group._id);
```

Comment:
Makes more sense to me if this uses the ID because you're not doing anything with the name and the ID should be the canonical way to refer to a group.

Reply:
I don't know how to handle this, as the only info that can be extracted from the view is the group name...

Reply:
See other response.